### ISSUE TITLE
projector  can not generate a image same with the target image.

### ISSUE THREAD

**Issue by tengshaofeng** [NVlabs/stylegan2-ada-pytorch#247]
@yulonglin @hkhailee @Gass2109 @nurpax 
you really provide many useful information to me. But When I load weights from "https://nvlabs-fi-cdn.nvidia.com/stylegan2-ada-pytorch/pretrained/transfer-learning-source-nets/ffhq-res512-mirror-stylegan2-noaug.pkl",
and run the projector.py ,The synth_image is different from target image. Why? I need your help.
target image:

![target](https://user-images.githubusercontent.com/9246739/173550884-9ed0a955-29b5-42ff-9f9a-2fefe43f38dd.png)
synth_image:

![proj](https://user-images.githubusercontent.com/9246739/173550985-c2f1c8f0-903a-4782-8688-270cc172a865.png)
# train loss:
step  994/1000: dist 0.22 loss 0.22 
step  995/1000: dist 0.22 loss 0.22 
step  996/1000: dist 0.22 loss 0.22 
step  997/1000: dist 0.22 loss 0.22 
step  998/1000: dist 0.22 loss 0.22 
step  999/1000: dist 0.22 loss 0.22 
step 1000/1000: dist 0.22 loss 0.22 
Elapsed: 75.3 s

**Comment by JackWolftoMc** [comment]
Of course there're gaps here

**Comment by tengshaofeng** [comment]
@JackWolftoMc Thanks for your reply.

**Comment by 1zong2** [comment]
Alignment is important for stylegan models. 
Align the image before project it.

**Comment by TimRuanwx** [comment]
> Alignment is important for stylegan models. Align the image before project it.

Thank you so much. The generated face seem to have unchangeable position and size.  
BTW: Is there any code for alignment in this repo?

**Comment by webdevserv** [comment]
No but you can import dlib to fid eyes.

detector = dlib.get_frontal_face_detector()
predictor = dlib.shape_predictor('shape_predictor_5_face_landmarks.dat')

**Comment by LisaLy123456** [comment]
> 感谢您的回复。

Does your problem has been solved? I encountered the same problem and I aligned the face, but the problem still exists.
